In [1]:
import pandas as pd

In [2]:
original_data = pd.read_csv("D:/user/ZengQ/07_Python_File/2022年销售提货_UTF-8.csv")

In [3]:
original_data.sample(10)

提货日期             收货客户              钢卷号               合同号    钢种  表面  \
125905  202211      无锡嘉德不锈钢有限公司  FJ2211050227-02     FJYJ22Q-00817  316L  2B   
131104  202211  昕宸安（上海）金属材料有限公司  FJ2211011071-01     FJYJ22Q-00878   304  2B   
143449  202212     佛山市鑫普辉贸易有限公司  FJ2212013580-04  FJYJ221208-Q-105   304  2B   
108004  202209    无锡青拓不锈钢制品有限公司  FJ2209590004-01     FJYJ22Q-00540  317L  2B   
129208  202211     佛山市鑫普辉贸易有限公司  FJ2211011931-03   FJYJ221110-Q-93   304  2B   
45317   202204     佛山新展金属材料有限公司  FJ2204011658-03     FJYJ22Q-00141   304  2B   
43239   202204       江苏青通贸易有限公司  FJ2204012545-02   FJYJ22Q-00215-B   304  2B   
33412   202204     张家港市安达贸易有限公司  FJ2203120002-02  FJYJ220331-E-016   301  2B   
102077  202209      佛山瑞钢达贸易有限公司  FJ2209050124-02     FJYJ22Q-00650  316L  2B   
126875  202211      福建青旭不锈钢有限公司  FJ2210013325-02     FJYJ22Q-00841   304  2B   

             实际规格  出库钢卷重量    含税单价       含税金额  产品用途码 质量等级     发货仓库   酸洗线  
125905  0.46*1250  11,460  29.320  336007.20   316L  一级品  国内贸易成品库  3#冷酸  
131104  0.80*1249  19,719  17.130  337786.47    304  一级品  国内贸易无锡库  2#冷酸  
143449  0.92*1251     190  14.600    2774.00    304  二级品  国内贸易成品库  2#冷酸  
108004  1.46*1252  10,802  35.765  386333.53   317L  一级品  国内贸易无锡库  3#冷酸  
129208  1.44*1253     397  14.600    5796.20    304  二级品  国内贸易成品库  2#冷酸  
45317   0.38*1247   7,683  20.520  157655.16  304DQ  一级品  国内贸易佛山库  1#冷酸  
43239   0.96*1249  11,025  20.420  225130.50    304  一级品  国内贸易无锡库  2#冷酸  
33412   0.84*1255     125  13.100    1637.50  301SP  二级品  国内贸易成品库  3#冷酸  
102077  0.70*1253  10,918  25.600  279500.80   316L  一级品  国内贸易佛山库  3#冷酸  
126875  0.94*1253  21,763  17.750  386293.25  304DQ  一级品  国内贸易成品库  2#冷酸

In [4]:
original_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 143754 entries, 0 to 143753
Data columns (total 14 columns):
 #   Column  Non-Null Count   Dtype  
---  ------  --------------   -----  
 0   提货日期    143754 non-null  int64  
 1   收货客户    143754 non-null  object 
 2   钢卷号     143754 non-null  object 
 3   合同号     143754 non-null  object 
 4   钢种      143754 non-null  object 
 5   表面      143754 non-null  object 
 6   实际规格    143754 non-null  object 
 7   出库钢卷重量  143754 non-null  object 
 8   含税单价    143754 non-null  float64
 9   含税金额    143754 non-null  float64
 10  产品用途码   143754 non-null  object 
 11  质量等级    143754 non-null  object 
 12  发货仓库    143754 non-null  object 
 13  酸洗线     141115 non-null  object 
dtypes: float64(2), int64(1), object(11)
memory usage: 15.4+ MB


一共有143754条数据。`提货日期`需要改为日期时间(转换后显示错误)，`出库钢卷重量`需要改为浮点数，`酸洗线`有空缺值

In [5]:
original_data[original_data["酸洗线"].isnull()]

提货日期           收货客户              钢卷号            合同号    钢种  表面  \
0       202201  浙江青清不锈钢科技有限公司  FJ2112112430-02  FJYJ21Q-00801   304  2B   
1       202201  浙江青清不锈钢科技有限公司  FJ2112112843-02  FJYJ21Q-00852   304  2B   
2       202201  浙江青清不锈钢科技有限公司  FJ2112112969-01  FJYJ21Q-00852   304  2B   
3       202201  浙江青清不锈钢科技有限公司  FJ2112140243-01  FJYJ21Q-00874  316L  2B   
4       202201  浙江青清不锈钢科技有限公司  FJ2112140243-02  FJYJ21Q-00874  316L  2B   
...        ...            ...              ...            ...   ...  ..   
81449   202207  江苏普鲁士金属制品有限公司  FJ2112110945-01  FJYJ22Q-00481   304  2B   
122961  202211    无锡盛宝不锈钢有限公司     FJ2210013421  FJYJ22Q-00830   304  2G   
123817  202211    无锡盛宝不锈钢有限公司     FJ2210013422  FJYJ22Q-00830   304  2G   
127670  202211    无锡盛宝不锈钢有限公司     FJ2210013423  FJYJ22Q-00830   304  2G   
131640  202211    无锡盛宝不锈钢有限公司     FJ2210013400  FJYJ22Q-00830   304  2G   

             实际规格  出库钢卷重量   含税单价       含税金额   产品用途码 质量等级     发货仓库  酸洗线  
0       0.82*1251  11,049  18.82  207942.18     304  一级品  国内贸易成品库  NaN  
1       0.65*1247  11,045  17.22  190194.90     304  一级品  国内贸易成品库  NaN  
2       0.92*1247  11,074  17.02  188479.48     304  一级品  国内贸易成品库  NaN  
3       0.38*1248  10,769  27.15  292378.35  316LCQ  一级品  国内贸易成品库  NaN  
4       0.38*1248  10,707  27.15  290695.05  316LCQ  一级品  国内贸易成品库  NaN  
...           ...     ...    ...        ...     ...  ...      ...  ...  
81449   0.82*1246   2,695  17.22   46407.90     304  一级品  国内贸易成品库  NaN  
122961  0.79*1245  22,569  17.95  405113.55   304M8  一级品  国内贸易成品库  NaN  
123817  0.89*1245  22,273  17.95  399800.35   304M8  一级品  国内贸易成品库  NaN  
127670  1.19*1245  22,083  17.95  396389.85   304M8  一级品  国内贸易成品库  NaN  
131640  1.79*1245  22,161  17.95  397789.95   304M8  一级品  国内贸易成品库  NaN  

[2639 rows x 14 columns]

检索后发现`酸洗线`有空缺值不影响销售统计

In [23]:
name = pd.Series({"青菜":"2214,1","白萝卜":"4562,2","西红柿":"7655,3","土豆":"3573,7","黄瓜":"2896,8"})
name.str.replace("2214,1","2214")
#print(name1)

青菜       2214
白萝卜    4562,2
西红柿    7655,3
土豆     3573,7
黄瓜     2896,8
dtype: object

In [35]:
#先修改数据类型，再进行统计审查
#original_data["提货日期"] = pd.to_datetime(original_data["提货日期"])
#original_data["出库钢卷重量"] = original_data["出库钢卷重量"].astype(float)
#删除"出库钢卷重量"里的','
original_data["出库钢卷重量"] = original_data["出库钢卷重量"].str.replace(",","")

In [36]:
original_data["出库钢卷重量"] = original_data["出库钢卷重量"].astype(float)
original_data["出库钢卷重量"]

0         11049.0
1         11045.0
2         11074.0
3         10769.0
4         10707.0
           ...   
143749      207.0
143750      172.0
143751      145.0
143752      168.0
143753      135.0
Name: 出库钢卷重量, Length: 143754, dtype: float64

In [37]:
original_data.describe()

提货日期         出库钢卷重量           含税单价           含税金额
count  143754.000000  143754.000000  143754.000000  143754.000000
mean   202206.583497    6448.215173      18.068023  124534.601878
std         3.318528    5725.235252       3.420119  118731.079216
min    202201.000000      93.000000       7.320000    1040.000000
25%    202204.000000     240.000000      16.370000    3932.400000
50%    202206.000000    7268.000000      17.400000  131916.340000
75%    202209.000000   10840.000000      19.420000  192685.000000
max    202212.000000   24286.000000      36.535000  730988.160000

数据初步检查没有问题  
还需要检查重复数据`钢卷号`  

In [44]:
original_data.duplicated(subset=["钢卷号"]).sum()

0

`发货仓库`和`质量等级`是否有不同名称代指同一地点和等级

In [45]:
original_data["发货仓库"].value_counts()

发货仓库
国内贸易成品库     93238
国内贸易佛山库     36133
国内贸易无锡库     12935
国内贸易福建外仓     1448
Name: count, dtype: int64

In [46]:
original_data["质量等级"].value_counts()

质量等级
一级品    88163
二级品    55155
合格品      419
三级品       17
Name: count, dtype: int64